webapp/channels/src/components/root/root.tsx

```diff
@@ -352,20 +313,26 @@ export default class Root extends React.PureComponent<Props, State> {
         if (!deepEqual(prevProps.theme, this.props.theme)) {
             Utils.applyTheme(this.props.theme);
         }
+
         if (this.props.location.pathname === '/') {
             if (this.props.noAccounts) {
                 prevProps.history.push('/signup_user_complete');
             } else if (this.props.showTermsOfService) {
                 prevProps.history.push('/terms_of_service');
             }
         }
+
         if (
             this.props.shouldShowAppBar !== prevProps.shouldShowAppBar ||
             this.props.rhsIsOpen !== prevProps.rhsIsOpen ||
             this.props.rhsIsExpanded !== prevProps.rhsIsExpanded
         ) {
             this.setRootMeta();
         }
+
+        if (!prevProps.isConfigLoaded && this.props.isConfigLoaded) {
+            this.setRudderConfig();
+        }
```

Comment:
Great change. That also makes it easy for us to eventually refactor this to pull all the Rudder logic into its own self-contained component